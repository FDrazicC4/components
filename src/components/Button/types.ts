import React from 'react';
import { Colors as GenericColors, StyledParentComponent } from '@components/types';


export type Variants = (
    | 'standard'
    | 'text'
);


export type Colors = (
    | GenericColors
    | 'link'
);


export type Sizes = (
    | 'small'
    | 'medium'
    | 'large'
);


export interface Props extends StyledParentComponent, React.ButtonHTMLAttributes<HTMLButtonElement> {

    /**
     * Fires when button is clicked.
     */
    onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void,

    /**
     * Determines the display style of the button
     * @default 'standard'
     */
    variant?: Variants,

    /**
     * The color of the button.
     * @default 'primary'
     */
    color?: Colors,

    /**
     * The size of the button.
     * @default 'medium'
     */
    size?: Sizes,

    /**
     * If `true`, will disable the button.
     * @default false
     */
    isDisabled?: boolean,

    /**
     * If `true`, the button will be disabled and show a spinner.
     * @default false
     */
    isLoading?: boolean,

    /**
     * Reduces the padding of the button.
     * @default false
     */
    isCompact?: boolean,

    /**
     * Makes the button a full-width block element.
     * @default false
     */
    isFluid?: boolean,

    /**
     * Rounds the corners of the button to make a circle.
     * @default false
     */
    isCircular?: boolean,

    /**
     * The type of the button.
     * @default 'button'
     */
    type?: 'button' | 'submit' | 'reset',

    /**
     * JSX (preferably an Icon) to be injected before the children.
     */
    startIcon?: React.ReactNode,

    /**
     * JSX (preferably an Icon) to be injected after the children.
     */
    endIcon?: React.ReactNode,

    /**
     * Determines the component that is used to render. Accepts any React component, custom and HTML elements.
     */
    component?: React.ElementType,

}
