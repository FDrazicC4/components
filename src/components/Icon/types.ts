import React from 'react';
import { StyledParentComponent, Colors as GenericColors } from '@components/types';


export type Colors = GenericColors | 'inherit';
export type Sizes = 'inherit' | 'small' | 'large' | 'default';


export interface Props extends StyledParentComponent {

    /**
     * Determines the color of the rendered Icon.
     * @default 'inherit'
     */
    color?: Colors,

    /**
     * If set, forces the `Icon` to render at the specified size.
     * @default 'inherit'
     */
    size?: Sizes,

}


export interface IconComponent extends React.ForwardRefExoticComponent<any> {
    Svg?: any,
}
