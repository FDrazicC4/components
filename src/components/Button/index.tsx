import React from 'react';
import clsx from 'clsx';
import { Button as MuiButton } from '@material-ui/core';
import Typography from '@components/Typography';
import Progress from '@components/Progress';
import { Props } from './types';


/**
 * The `Button` component represents a clickable button, which can be used in forms or anywhere in the application
 * that needs simple, standard button functionality.
 */
export const Button = React.forwardRef(({
    variant = 'standard',
    color = 'primary',
    size = 'medium',
    isLoading = false,
    isCompact = false,
    isCircular = false,
    isFluid = false,
    isDisabled = false,
    startIcon,
    endIcon,
    children,
    className,
    ...otherProps
}: Props, ref: React.Ref<any>): React.ReactElement => {


    const buttonClasses = clsx(
        'brew-Button',
        `brew-Button--variant-${variant}`,
        `brew-Button--color-${color}`,
        `brew-Button--size-${size}`,
        { 'brew-Button--isLoading': isLoading },
        { 'brew-Button--isDisabled': isDisabled || isLoading },
        { 'brew-Button--isCompact': isCompact },
        { 'brew-Button--isCircular': isCircular && !isFluid },
        { 'brew-Button--isFluid': isFluid },
        className,
    );


    return (
        <MuiButton
            className={buttonClasses}
            disabled={isDisabled || isLoading}
            ref={ref}
            type="button"
            {...otherProps}
        >
            <div className="brew-Button__content">
                {startIcon}
                {children && (
                    <Typography className="brew-Button__text" variant="button">
                        {children}
                    </Typography>
                )}
                {endIcon}
            </div>
            <div className="brew-Button__loadingIndicator">
                <Progress.Circular color={color === 'link' ? 'primary' : color} />
            </div>
        </MuiButton>
    );


});


Button.displayName = 'Button';


export default Button;
