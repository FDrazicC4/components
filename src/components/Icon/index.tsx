import React from 'react';
import MuiIcon from '@material-ui/core/Icon';
import SvgIcon from './components/SvgIcon';
import { IconComponent, Props } from './types';


/**
 * The `Icon` component will display an icon from any icon font that supports ligatures. As a prerequisite, you must
 * include one, such as the Material icon font in your project, for instance, via Google Web Fonts:
 *
 * ```
 * <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons" />
 * ```
 *
 * Icon will set the correct class name for the Material icon font. For other fonts, you must supply the class name
 * using the Icon component's `className` property.
 *
 * To use an icon simply wrap the icon name (font ligature) with the `Icon` component, for example: `<Icon>star</Icon>`.
 */
export const Icon: IconComponent = React.forwardRef(({
    color = 'inherit',
    size = 'inherit',
    ...otherProps
}: Props, ref: React.Ref<any>): React.ReactElement => {


    const classes = {
        root: `brew-Icon brew-Icon--color-${color} brew-Icon--size-${size}`,
    };


    return (
        <MuiIcon
            classes={classes}
            component="i"
            fontSize={size}
            ref={ref}
            {...otherProps}
        />
    );


});


Icon.Svg = SvgIcon;


Icon.displayName = 'Icon';


export default Icon;
