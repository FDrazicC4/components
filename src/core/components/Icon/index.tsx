import React from 'react';
import clsx from 'clsx';
import { Props } from './types';
import styles from './styles.scss';


function Icon(props: Props) {


    const {
        children,
        color = 'primary',
        size = 'inherit',
        className,
        ...otherProps
    } = props;


    /**
     * Combine our classes based on props
     */
    const classes = clsx(
        styles.wrapper,
        styles[`color--${color}`],
        styles[`size--${size}`],
        className,
    );


    /** Lazy load our icon */
    const Image = React.lazy(() => {
        return new Promise((resolve) => {
            import(`./material-icons/dist/${children}.js`)
                .then(response => resolve(response));
        });
    });


    return (
        <span
            className={classes}
            {...otherProps}
        >
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="100%">
                <React.Suspense fallback="&#9744;">
                    <Image />
                </React.Suspense>
            </svg>
        </span>
    );


}


export default Icon;
