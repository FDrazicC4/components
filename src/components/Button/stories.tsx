import React from 'react';
import { text, select, boolean } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import { ArrowBack, AddCircleOutline } from '@material-ui/icons';
import Button from './index';
import { Variants, Colors, Sizes } from './types';
import Flag from '@components/Flag';


export default {
    component: Button,
    parameters: {
        componentSubtitle: <Flag color="success">Stable</Flag>,
    },
    title: 'Stable|User Input/Button',
};


export const Sandbox = (): React.ReactElement => {
    const children: React.ReactNode = text('children', 'Click me!');
    const variant: Variants = select('variant', ['standard', 'text'], 'standard');
    const color: Colors = select('color', [
        'primary',
        'secondary',
        'success',
        'error',
        'warning',
        'info',
        'light',
        'dark',
        'link',
    ], 'primary');
    const size: Sizes = select('size', ['small', 'medium', 'large'], 'medium');
    const isLoading: boolean = boolean('isLoading', false);
    const isCompact: boolean = boolean('isCompact', false);
    const isFluid: boolean = boolean('isFluid', false);
    const isCircular: boolean = boolean('isCircular', false);
    const isDisabled: boolean = boolean('isDisabled', false);
    return (
        <Button
            color={color}
            isCircular={isCircular}
            isCompact={isCompact}
            isDisabled={isDisabled}
            isFluid={isFluid}
            isLoading={isLoading}
            onClick={action('button-click')}
            size={size}
            variant={variant}
        >
            {children}
        </Button>
    );
};


export const General = (): React.ReactElement => <Button>Default</Button>;
export const Standard = (): React.ReactElement => <Button variant="standard">Click me!</Button>;
export const Text = (): React.ReactElement => <Button variant="text">Click me!</Button>;
export const WithStartIcon = (): React.ReactElement => <Button startIcon={<ArrowBack />}>Go Back</Button>;
export const WithEndIcon = (): React.ReactElement => <Button endIcon={<AddCircleOutline />}>Add Account</Button>;
export const OnlyIcon = (): React.ReactElement => <Button endIcon={<AddCircleOutline />} />;
export const Loading = (): React.ReactElement => <Button isLoading variant="standard">Click me!</Button>;
export const Fluid = (): React.ReactElement => <Button isFluid variant="standard">Click me!</Button>;
export const Compact = (): React.ReactElement => <Button isCompact variant="standard">Click me!</Button>;
export const Disabled = (): React.ReactElement => <Button isDisabled variant="standard">Click me!</Button>;
export const Link = (): React.ReactElement => <Button href="http://google.com">Go to Google</Button>;
