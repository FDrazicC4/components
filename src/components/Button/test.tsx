import React from 'react';
import renderer from 'react-test-renderer';
import Button from '@components/Button';
import Icon from '@components/Icon';


it('renders correctly', () => {
    const tree = renderer
        .create(<Button>Foo</Button>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});


it('renders correctly with various props', () => {
    const tree = renderer
        .create((
            <React.Fragment>
                <Button startIcon={<Icon>arrow_back</Icon>}>Go Back</Button>
                <Button variant="standard">Click me!</Button>
                <Button variant="text">Click me!</Button>
                <Button endIcon={<Icon>add_circle_outline</Icon>}>Add Account</Button>
                <Button endIcon={<Icon>add_circle_outline</Icon>} />
                <Button isLoading variant="standard">Click me!</Button>
                <Button isFluid variant="standard">Click me!</Button>
                <Button isCompact variant="standard">Click me!</Button>
                <Button isDisabled variant="standard">Click me!</Button>
            </React.Fragment>
        ))
        .toJSON();
    expect(tree).toMatchSnapshot();
});
