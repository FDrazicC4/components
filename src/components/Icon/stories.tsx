import React from 'react';
import { select } from '@storybook/addon-knobs';
import Icon from './index';
import Flag from '@components/Flag';
import { Colors as ColorOptions, Sizes as SizeOptions } from './types';


export default {
    component: Icon,
    parameters: {
        componentSubtitle: <Flag color="success">Stable</Flag>,
    },
    title: 'Stable|Data Display/Icons/Icon',
};


export const Sandbox = (): React.ReactElement => {

    const color: ColorOptions = select('color', [
        'primary',
        'secondary',
        'success',
        'error',
        'warning',
        'info',
        'light',
        'dark',
        'inherit',
    ], 'inherit');
    const size: SizeOptions = select('size', [
        'small',
        'default',
        'large',
        'inherit',
    ], 'inherit');

    return (
        <Icon color={color} size={size}>add_circle</Icon>
    );

};


export const Default = (): React.ReactElement => (
    <Icon>add_circle</Icon>
);


export const Sizes = (): React.ReactElement => (
    <React.Fragment>
        <Icon>add_circle</Icon>
        <Icon size="small">add_circle</Icon>
        <Icon size="default">add_circle</Icon>
        <Icon size="large">add_circle</Icon>
        <Icon size="inherit">add_circle</Icon>
    </React.Fragment>
);


export const Colors = (): React.ReactElement => (
    <React.Fragment>
        <Icon>add_circle</Icon>
        <Icon color="primary">add_circle</Icon>
        <Icon color="secondary">add_circle</Icon>
        <Icon color="success">add_circle</Icon>
        <Icon color="warning">add_circle</Icon>
        <Icon color="error">add_circle</Icon>
        <Icon color="info">add_circle</Icon>
        <Icon color="dark">add_circle</Icon>
        <Icon color="light">add_circle</Icon>
        <Icon color="inherit">add_circle</Icon>
    </React.Fragment>
);
