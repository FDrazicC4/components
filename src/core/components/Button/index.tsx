import React from 'react';
import clsx from 'clsx';
import { Flipper, Flipped } from 'react-flip-toolkit';
import Typography from '../Typography';
import Icon from '../Icon';
import Progress from '../Progress';
import { Props } from './types';
import styles from './styles.scss';


/**
 * The `Button` component represents a clickable button, which can be used in forms or anywhere in the application
 * that needs simple, standard button functionality.
 */
function Button({
    variant = 'standard',
    color = 'primary',
    size = 'medium',
    isLoading = false,
    isCompact = false,
    isCircular = false,
    isFluid = false,
    disabled = false,
    children,
    className,
    ...otherProps
}: Props) {


    const buttonClasses = clsx(
        styles.wrapper,
        styles[`variant--${variant}`],
        styles[`color--${color}`],
        styles[`size--${size}`],
        { [styles.isLoading]: isLoading },
        { [styles.disabled]: disabled || isLoading },
        { [styles.isCompact]: isCompact },
        { [styles.isCircular]: isCircular && !isFluid },
        { [styles.isFluid]: isFluid },
        className,
    );


    return (
        <Flipper flipKey={JSON.stringify([isLoading, variant, color, size])}>
            <Flipped flipId="wrapper">
                <button
                    className={buttonClasses}
                    type="button"
                    disabled={disabled || isLoading}
                    aria-label={(variant === 'icon' && typeof children === 'string') ? children : undefined}
                    {...otherProps}
                >
                    <Flipped flipId="content">
                        <div className={styles.content}>
                            <span className={styles.buttonText}>
                                <Typography>{children}</Typography>
                            </span>
                            {variant === 'icon' && typeof children === 'string' &&
                                <span className={styles.icon}>
                                    <Icon color="inherit">{children}</Icon>
                                </span>
                            }
                        </div>
                    </Flipped>
                    <Flipped flipId="loader">
                        <div className={styles.loadingIndicator}>
                            <Progress variant="circular" color={color} />
                        </div>
                    </Flipped>
                </button>
            </Flipped>
        </Flipper>
    );


}


export default Button;
