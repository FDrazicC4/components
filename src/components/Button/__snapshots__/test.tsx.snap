// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text brew-Button brew-Button--variant-standard brew-Button--color-primary brew-Button--size-medium"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    <div
      className="brew-Button__content"
    >
      <span
        className="MuiTypography-root brew-Typography brew-Typography--display-initial brew-Typography--variant-button brew-Typography--align-inherit brew-Button__text MuiTypography-button"
      >
        Foo
      </span>
    </div>
    <div
      className="brew-Button__loadingIndicator"
    >
      <div
        className="MuiCircularProgress-root brew-ProgressCircular--variant-indeterminate brew-ProgressCircular--color-primary brew-ProgressCircular--size-inherit MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={20}
            strokeWidth={4}
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </span>
</button>
`;

exports[`renders correctly with various props 1`] = `
Array [
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text brew-Button brew-Button--variant-standard brew-Button--color-primary brew-Button--size-medium"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <div
        className="brew-Button__content"
      >
        <i
          className="brew-Icon material-icons brew-Icon--color-inherit"
        >
          arrow_back
        </i>
        <span
          className="MuiTypography-root brew-Typography brew-Typography--display-initial brew-Typography--variant-button brew-Typography--align-inherit brew-Button__text MuiTypography-button"
        >
          Go Back
        </span>
      </div>
      <div
        className="brew-Button__loadingIndicator"
      >
        <div
          className="MuiCircularProgress-root brew-ProgressCircular--variant-indeterminate brew-ProgressCircular--color-primary brew-ProgressCircular--size-inherit MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style={
            Object {
              "height": 40,
              "width": 40,
            }
          }
        >
          <svg
            className="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx={44}
              cy={44}
              fill="none"
              r={20}
              strokeWidth={4}
              style={Object {}}
            />
          </svg>
        </div>
      </div>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>,
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text brew-Button brew-Button--variant-standard brew-Button--color-primary brew-Button--size-medium"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <div
        className="brew-Button__content"
      >
        <span
          className="MuiTypography-root brew-Typography brew-Typography--display-initial brew-Typography--variant-button brew-Typography--align-inherit brew-Button__text MuiTypography-button"
        >
          Click me!
        </span>
      </div>
      <div
        className="brew-Button__loadingIndicator"
      >
        <div
          className="MuiCircularProgress-root brew-ProgressCircular--variant-indeterminate brew-ProgressCircular--color-primary brew-ProgressCircular--size-inherit MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style={
            Object {
              "height": 40,
              "width": 40,
            }
          }
        >
          <svg
            className="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx={44}
              cy={44}
              fill="none"
              r={20}
              strokeWidth={4}
              style={Object {}}
            />
          </svg>
        </div>
      </div>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>,
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text brew-Button brew-Button--variant-text brew-Button--color-primary brew-Button--size-medium"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <div
        className="brew-Button__content"
      >
        <span
          className="MuiTypography-root brew-Typography brew-Typography--display-initial brew-Typography--variant-button brew-Typography--align-inherit brew-Button__text MuiTypography-button"
        >
          Click me!
        </span>
      </div>
      <div
        className="brew-Button__loadingIndicator"
      >
        <div
          className="MuiCircularProgress-root brew-ProgressCircular--variant-indeterminate brew-ProgressCircular--color-primary brew-ProgressCircular--size-inherit MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style={
            Object {
              "height": 40,
              "width": 40,
            }
          }
        >
          <svg
            className="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx={44}
              cy={44}
              fill="none"
              r={20}
              strokeWidth={4}
              style={Object {}}
            />
          </svg>
        </div>
      </div>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>,
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text brew-Button brew-Button--variant-standard brew-Button--color-primary brew-Button--size-medium"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <div
        className="brew-Button__content"
      >
        <span
          className="MuiTypography-root brew-Typography brew-Typography--display-initial brew-Typography--variant-button brew-Typography--align-inherit brew-Button__text MuiTypography-button"
        >
          Add Account
        </span>
        <i
          className="brew-Icon material-icons brew-Icon--color-inherit"
        >
          add_circle_outline
        </i>
      </div>
      <div
        className="brew-Button__loadingIndicator"
      >
        <div
          className="MuiCircularProgress-root brew-ProgressCircular--variant-indeterminate brew-ProgressCircular--color-primary brew-ProgressCircular--size-inherit MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style={
            Object {
              "height": 40,
              "width": 40,
            }
          }
        >
          <svg
            className="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx={44}
              cy={44}
              fill="none"
              r={20}
              strokeWidth={4}
              style={Object {}}
            />
          </svg>
        </div>
      </div>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>,
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text brew-Button brew-Button--variant-standard brew-Button--color-primary brew-Button--size-medium"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <div
        className="brew-Button__content"
      >
        <i
          className="brew-Icon material-icons brew-Icon--color-inherit"
        >
          add_circle_outline
        </i>
      </div>
      <div
        className="brew-Button__loadingIndicator"
      >
        <div
          className="MuiCircularProgress-root brew-ProgressCircular--variant-indeterminate brew-ProgressCircular--color-primary brew-ProgressCircular--size-inherit MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style={
            Object {
              "height": 40,
              "width": 40,
            }
          }
        >
          <svg
            className="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx={44}
              cy={44}
              fill="none"
              r={20}
              strokeWidth={4}
              style={Object {}}
            />
          </svg>
        </div>
      </div>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>,
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text brew-Button brew-Button--variant-standard brew-Button--color-primary brew-Button--size-medium brew-Button--isLoading brew-Button--isDisabled Mui-disabled Mui-disabled"
    disabled={true}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={-1}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <div
        className="brew-Button__content"
      >
        <span
          className="MuiTypography-root brew-Typography brew-Typography--display-initial brew-Typography--variant-button brew-Typography--align-inherit brew-Button__text MuiTypography-button"
        >
          Click me!
        </span>
      </div>
      <div
        className="brew-Button__loadingIndicator"
      >
        <div
          className="MuiCircularProgress-root brew-ProgressCircular--variant-indeterminate brew-ProgressCircular--color-primary brew-ProgressCircular--size-inherit MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style={
            Object {
              "height": 40,
              "width": 40,
            }
          }
        >
          <svg
            className="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx={44}
              cy={44}
              fill="none"
              r={20}
              strokeWidth={4}
              style={Object {}}
            />
          </svg>
        </div>
      </div>
    </span>
  </button>,
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text brew-Button brew-Button--variant-standard brew-Button--color-primary brew-Button--size-medium brew-Button--isFluid"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <div
        className="brew-Button__content"
      >
        <span
          className="MuiTypography-root brew-Typography brew-Typography--display-initial brew-Typography--variant-button brew-Typography--align-inherit brew-Button__text MuiTypography-button"
        >
          Click me!
        </span>
      </div>
      <div
        className="brew-Button__loadingIndicator"
      >
        <div
          className="MuiCircularProgress-root brew-ProgressCircular--variant-indeterminate brew-ProgressCircular--color-primary brew-ProgressCircular--size-inherit MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style={
            Object {
              "height": 40,
              "width": 40,
            }
          }
        >
          <svg
            className="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx={44}
              cy={44}
              fill="none"
              r={20}
              strokeWidth={4}
              style={Object {}}
            />
          </svg>
        </div>
      </div>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>,
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text brew-Button brew-Button--variant-standard brew-Button--color-primary brew-Button--size-medium brew-Button--isCompact"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <div
        className="brew-Button__content"
      >
        <span
          className="MuiTypography-root brew-Typography brew-Typography--display-initial brew-Typography--variant-button brew-Typography--align-inherit brew-Button__text MuiTypography-button"
        >
          Click me!
        </span>
      </div>
      <div
        className="brew-Button__loadingIndicator"
      >
        <div
          className="MuiCircularProgress-root brew-ProgressCircular--variant-indeterminate brew-ProgressCircular--color-primary brew-ProgressCircular--size-inherit MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style={
            Object {
              "height": 40,
              "width": 40,
            }
          }
        >
          <svg
            className="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx={44}
              cy={44}
              fill="none"
              r={20}
              strokeWidth={4}
              style={Object {}}
            />
          </svg>
        </div>
      </div>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>,
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text brew-Button brew-Button--variant-standard brew-Button--color-primary brew-Button--size-medium brew-Button--isDisabled Mui-disabled Mui-disabled"
    disabled={true}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={-1}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <div
        className="brew-Button__content"
      >
        <span
          className="MuiTypography-root brew-Typography brew-Typography--display-initial brew-Typography--variant-button brew-Typography--align-inherit brew-Button__text MuiTypography-button"
        >
          Click me!
        </span>
      </div>
      <div
        className="brew-Button__loadingIndicator"
      >
        <div
          className="MuiCircularProgress-root brew-ProgressCircular--variant-indeterminate brew-ProgressCircular--color-primary brew-ProgressCircular--size-inherit MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style={
            Object {
              "height": 40,
              "width": 40,
            }
          }
        >
          <svg
            className="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx={44}
              cy={44}
              fill="none"
              r={20}
              strokeWidth={4}
              style={Object {}}
            />
          </svg>
        </div>
      </div>
    </span>
  </button>,
]
`;
